### ISSUE TITLE
Upgrade project to Vue 3.x

### ISSUE THREAD

**Issue by brylie** [creativecommons/chooser#356]
Vue 3.0 was [released in September 2020](https://github.com/vuejs/vue-next/releases/tag/v3.0.0). However, this project is still using Vue 2.x.

Decide when it is timely to upgrade this project to Vue 3.x

## Description
One of the design constraints of Vue 3.x was to preserve backwards-compatibility with Vue 2.x projects. This would ideally mean very little would need to change when the Chooser project upgrades to Vue 3.x.

## Alternatives
We can stay at 2.x, but that would mean decreasing support as 2.x reaches end-of-life. 

## Additional context

The upcoming Vue 3.1 release includes a "[Migration Build" to help projects transition from 2.x to 3.x](https://github.com/vuejs/vue-next/tree/master/packages/vue-compat).

The Vue User Guide is defaulting to 3.x to signal that projects should be targeting the 3.x release.

## Implementation

- [ ] I would be interested in implementing this feature.

**Comment by obulat** [comment]
Very exciting change! :)

**Comment by sadn1ck** [comment]
(new here, 👋🏽)So, I was having a look at this locally, and I might be wrong, but wouldn't Vocabulary also need to be updated to vue 3 for chooser to be able to use the components from there? (v-radio, v-button, etc)

PS: I was able to get the build running on vue 3 but there were a lot of errors with component resolution, v-button css issues and i18n issues (vue-i18n v8 is for vue2 and v9 is for vue3)

**Comment by brylie** [comment]
@sadn1ck good point! :smiley: 

I've created an upstream issue to track the upgrade to Vue 3.x, which will enable this issue to move forward.

https://github.com/creativecommons/chooser/issues/360

By the way, are you available to help with some Vue development? We have a [pull request underway to simplify the repository structure](https://github.com/creativecommons/vocabulary/pull/933) of the CC Vocabulary monorepo and need a few more [changes to make sure all of the components will render](https://github.com/creativecommons/vocabulary/pull/933#issuecomment-884223802).

**Comment by sadn1ck** [comment]
Yep @brylie! I'd love to help out. I'm not familiar with storybook right now, so I'll have a look and start working on it as soon as I can.

**Comment by brylie** [comment]
OK, great! I think there is a basic pattern to follow to fix the remaining components. I will @ mention you on the related thread so we can pick up the conversation there. For what it's worth, I think we should finish the "simplify the project structure using Vue CLI" task before considering the Vue 3 upgrade. What do you think?

**Comment by sadn1ck** [comment]
Yep! Let's complete that first. That way we can focus on one thing at a time.

**Comment by possumbilities** [comment]
@ImaJin14 likely, but there is a larger refactor of Vocabulary happening that will likely result in correcting this issue downstream. Going to put this as blocked ti denote that.

**Comment by possumbilities** [comment]
Rendered obsolete by the new Chooser:
* https://github.com/creativecommons/chooser/pull/570

Closing.
